app/main.py

```diff
@@ -1 +1,34 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name:"
+                f" {self.name},"
+                f" Health: {self.health},"
+                f" Hidden: {self.hidden}}}"
+                )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Herbivore) -> None:
+        if isinstance(herbivore, Carnivore) or herbivore.hidden:
```

Comment:
You should accept only `Herbivore` for biting according to the task description, that is why you should directly check that. There could be another class added later that could impact the logic in its current form, so change it to 
```suggestion
        if not isinstance(herbivore, Herbivore) or herbivore.hidden:
```

Reply:
Все виправлено) дякую